TijanaSonjic/assignment1/Q2.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+
+template <class T>
+class List{
+private:
+	template <class T>
+	class Element {
+	public:
+		T info;
+		Element* next;
+	
+		Element(T info, Element* next = nullptr) {
+			this->info = info;
+			this->next = next;
+		}
+	};
+	Element<T>* head;
+	Element<T>* tail;
+	int count;
+public:
+	List() : head(nullptr), tail(nullptr), count(0){}
+
+	~List();
+
+	void addElement(T info);
+
+	Element<T>* getKthToLastElement(int position);
+
+	void printValueOfKthKthToLastElement(int position);
+
+	void printList();
+};
+
+
+template <typename T>
+void List<T>::addElement(T info) {
+	if (!head) {
+		head = tail = new Element<T>(info);
+	}
+	else {
+		tail->next = new Element<T>(info);
+		tail = tail->next;
+	}
+	count++;
+}
+
+template <class T>
+List<T>::Element<T>* List<T>::getKthToLastElement(int position) {
+	if (position < 0 || position >= count)
+		return nullptr;
+
+	// Make a distance between these two pointers for position elements
+	Element<T>* firstPointer = head;
+	Element<T>* secondPointer = head;
+	while (position >= 0) {
+		firstPointer = firstPointer->next;
+		position--;
+	}
+
+	// Move both pointers
+	while (firstPointer) {
+		firstPointer = firstPointer->next;
+		secondPointer = secondPointer->next;
+	}
+
+	return secondPointer;
+}

```

Comment:
Nothing to add here, solution is looks complete, short and clear to understand :)